crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -1766,33 +1644,18 @@ mod tests {
     }
 
     #[test]
-    fn test_key_filter_intersection() {
+    fn test_object_filter_intersection() {
         let i1 = SuiAddress::from_str("0x1").unwrap();
         let i2 = SuiAddress::from_str("0x2").unwrap();
         let i3 = SuiAddress::from_str("0x3").unwrap();
-        let i4 = SuiAddress::from_str("0x4").unwrap();
 
         let f0 = ObjectFilter {
             object_ids: Some(vec![i1, i3]),
-            object_keys: Some(vec![
```

Comment:
After this change, `f0` now equals `f1`, and further below `f3` is now a completely empty filter -- that suggests to me there is some scope to simplify these tests.

To fix these tests, you need to work from what the tests looked like before, understand the properties that were being tested, figure out which ones are still relevant after `objectKeys` have been removed and then write new (much simpler) tests to test those properties. Tests are an important form of documentation and as it stands these tests are quite confusing.